src/rdkafka_conf.c

```diff
@@ -1024,6 +1111,67 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
 	{ 0, /* End */ }
 };
 
+/**
+ * @returns the property object for \p name in \p scope, or NULL if not found.
+ * @remark does not work with interceptor configs.
+ */
+const struct rd_kafka_property *
+rd_kafka_conf_prop_find (int scope, const char *name) {
+        const struct rd_kafka_property *prop;
+
+ restart:
+        for (prop = rd_kafka_properties ; prop->name ; prop++) {
+
+                if (!(prop->scope & scope))
+                        continue;
+
+                if (strcmp(prop->name, name))
+                        continue;
+
+                if (prop->type == _RK_C_ALIAS) {
+                        /* Caller supplied an alias, restart
+                         * search for real name. */
+                        name = prop->sdef;
+                        goto restart;
+                }
+
+                return prop;
+        }
+
+        return NULL;
+}
+
+/**
+ * @returns true if property has been set/modified, else 0.
+ *          If \p name is unknown 0 is returned.
+ */
+static rd_bool_t rd_kafka_conf_is_modified (const rd_kafka_conf_t *conf,
+                                            const char *name) {
+        const struct rd_kafka_property *prop;
+
+        if (!(prop = rd_kafka_conf_prop_find(_RK_GLOBAL, name)))
+            return 0;
+
+        return rd_kafka_anyconf_is_modified(conf, prop);
+}
+
+
+/**
+ * @returns true if property has been set/modified, else 0.
```

Comment:
nit: probably should use `rd_false`, `rd_true`, or at least not switch between int and bool mid sentence.

Reply:
haha